### ISSUE TITLE
python: segfault from Future.get()

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#2045]
I was trying to be too lazy and ran across a segfault when running
```python
#!/usr/bin/env python
import json 
import flux
result = flux.Flux().rpc("kvs.stats.get").get()
print (json.dumps (result))
```
```console                                               
$ ./stats2.py                                  
Segmentation fault (core dumped)
```

This works though:
```python
#!/usr/bin/env python
import json
import flux
f = flux.Flux()
result = f.rpc("kvs.stats.get").get()
print (json.dumps (result))
```

```console
$ ./stats.py
{"cache": {"obj size total (MiB)": 0.00011920928955078125, "#faults": 0, "#obj dirty": 0, "#obj incomplete": 0, "obj size (KiB)": {"count": 1, "max": 0.125, "mean": 0.125, "stddev": 0.0, "min": 0.125}}, "namespace": {"primary": {"store revision": 3, "#transactions": 0, "#syncers": 0, "#no-op stores": 0, "#readytransactions": 0}}}
```

Stack trace:
```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fde6a40ada1 in flux_pollevents (h=<optimized out>) at handle.c:844
844	        if ((events = h->ops->pollevents (h->impl)) < 0)
(gdb) bt
#0  0x00007fde6a40ada1 in flux_pollevents (h=<optimized out>) at handle.c:844
#1  0x00007fde6a4181f1 in get_pollevents (h=<optimized out>) at ev_flux.c:25
#2  0x00007fde6a41830d in prepare_cb (loop=0x5614cdb85490, w=0x5614cdb85f18, 
    revents=<optimized out>) at ev_flux.c:40
#3  0x00007fde6a44dfc3 in ev_invoke_pending (loop=0x5614cdb85490) at ev.c:3322
#4  0x00007fde6a451834 in ev_run (loop=0x5614cdb85490, flags=0) at ev.c:3615
#5  0x00007fde6a40bf66 in flux_reactor_run (r=0x5614cdb0e110, 
    flags=flags@entry=0) at reactor.c:126
#6  0x00007fde6a419e73 in flux_future_wait_for (f=f@entry=0x5614cdb85e00, 
    timeout=timeout@entry=-1) at future.c:482
#7  0x00007fde6a41a0a1 in flux_future_get (f=0x5614cdb85e00, 
    result=result@entry=0x7fff144b54e0) at future.c:509
#8  0x00007fde6a415048 in flux_rpc_get (f=<optimized out>, s=0x5614cdbcf420)
    at rpc.c:108
#9  0x00007fde6a6f282a in _cffi_f_flux_rpc_get (self=<optimized out>, 
    args=<optimized out>) at _core.c:18147
#10 0x00005614cd57a8db in PyEval_EvalFrameEx ()
#11 0x00005614cd571d0a in PyEval_EvalCodeEx ()
#12 0x00005614cd58d8bc in ?? ()
#13 0x00005614cd5a60de in ?? ()
#14 0x00005614cd55d77e in PyObject_Call ()
#15 0x00005614cd6025f7 in ?? ()
#16 0x00005614cd55d77e in PyObject_Call ()
#17 0x00005614cd576a21 in PyEval_EvalFrameEx ()
#18 0x00005614cd571d0a in PyEval_EvalCodeEx ()
#19 0x00005614cd579c38 in PyEval_EvalFrameEx ()
#20 0x00005614cd579232 in PyEval_EvalFrameEx ()
#21 0x00005614cd579232 in PyEval_EvalFrameEx ()
#22 0x00005614cd571d0a in PyEval_EvalCodeEx ()
#23 0x00005614cd571629 in PyEval_EvalCode ()
#24 0x00005614cd5a261f in ?? ()
#25 0x00005614cd59d322 in PyRun_FileExFlags ()
#26 0x00005614cd59c67d in PyRun_SimpleFileExFlags ()
#27 0x00005614cd54b1ab in Py_Main ()
#28 0x00007fde6b9e3b97 in __libc_start_main (main=0x5614cd54ab10 <main>, 
    argc=2, argv=0x7fff144b6328, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7fff144b6318)
    at ../csu/libc-start.c:310
#29 0x00005614cd54aa2a in _start ()
```

**Comment by SteVwonder** [comment]
😱 Hmmmm.  My guess is that the `Flux` object is being garbage collected (which results in the handle being closed) in between the rpc creation and the `get`.   I ran into a similar problem with `then` callbacks on "anonymous" RPCs.  The RPC would be garbage collected before the callback was called.  For the callback, I just kept my own reference to the RPC until the callback ran. 

For this scenario, I'm not sure what the right solution is.  One thing that *might* work is to store a reference to the Flux object in the RPC object, which would hopefully prevent the Flux object from being GC'd while the RPC is still alive.  If that doesn't work, then idk.  Let me give that a try.
